.github/workflows/gradle.yml

```diff
@@ -24,31 +24,22 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Setup Gradle Dependencies Cache
-        uses: actions/cache@v3
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-caches-${{ hashFiles('**/*.gradle', '**/*.gradle.kts') }}
-
-      - name: Setup Gradle Wrapper Cache
-        uses: actions/cache@v3
-        with:
-          path: ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}
-
       - name: Set up JDK ${{ matrix.java-version }}
         uses: actions/setup-java@v3
         with:
           distribution: temurin
           java-version: ${{ matrix.java-version }}
 
-      - name: Run tests with Gradle
-        uses: eskatos/gradle-command-action@v1
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
         with:
-          arguments: |
-            test 
-            --stacktrace 
-            -Pkotlin.version=${{ matrix.kotlin-version }} 
+          cache-read-only: false
+
+      - name: Run tests with Gradle
+        run: |
+          ./gradlew check
```

Comment:
I also changed `./gradlew test` to `./gradlew check`, because I noticed that the API dumps are not up to date. The `apiCheck` task will run if the `check` task runs, not `test.